
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css">
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js"></script>
  </head>
<body ng-app="app">
<div class="container">
        <h2>Metascape Gene List Analysis Report</h2><p/>  <a href="http://metascape.org">metascape.org</a><sup>1</sup><p/>

              <h3>Heatmap Summary</h3>
              <div class="panel panel-info">
                <div class="panel-heading">Figure 1. Heatmap of enriched terms across input gene lists, colored by p-values.</div>
                <div class="panel-body">

                <table>
                    <tr>
                        <td>
                            <img src="./Enrichment_heatmap/HeatmapSelectedGO.png" style="width:1000px;">
                        </td>
                    </tr>
                    <tr>
                        <td align='center'>
                            
                <a href='./Enrichment_heatmap/HeatmapSelectedGO.pdf' title='download PDF file'>
                <img class='link' src='icon/PDF48.png' >
                </a>
                
                        </td>
                    </tr>
                    
                <tr><td>The top-level Gene Ontology biological processes can be viewed here.<br>
                <a href="./Enrichment_heatmap/HeatmapSelectedGOParent.png"><img src="./Enrichment_heatmap/HeatmapSelectedGOParent.png" style="height:auto; width:auto; max-height:100px;"></a>
                <a href='./Enrichment_heatmap/HeatmapSelectedGOParent.pdf' title='download PDF file'>
                <img class='link' src='icon/PDF48.png' >
                </a></td></tr>
                
                </table>
                </div>
              </div><p/>The heatmap can be interactively viewed using <a href="http://jtreeview.sourceforge.net">JTreeView</a><sup>2</sup> (.cdt, .gtr and .atr files can be found in the Zip package).<p/>
      <h3>Gene Lists</h3>
      User-provided gene identifiers are first converted into their corresponding A. thaliana Entrez gene IDs using the latest version of the database (last updated on 2024-05-01).  If multiple identifiers correspond to the same Entrez gene ID, they will be considered as a single Entrez gene ID in downstream analyses.   Each gene list is assigned a unique color, which is used throughout the analysis.   The gene lists are summarized in Table 1.<p/>
          <div class="panel panel-info">
            <div class="panel-heading">Table 1. Statistics of input gene lists.</div>
            <div class="panel-body"><TABLE class="table">
  <THEAD>
    <TR>
      <TH class="info">Name</TH>
      <TH class="info">Total</TH>
      <TH class="info">Unique</TH>
      <TH class="info">Color Code</TH>
    </TR>
  </THEAD>
  <TBODY>
    <TR>
      <TD>root_angle|5 OSDR|3 OSDR</TD>
      <TD>6</TD>
      <TD>6</TD>
      <TD><div style="background-color:#E41A1C !important;width:80px;height:20px;"></div></TD>
    </TR>
    <TR>
      <TD>tortosity|5 OSDR</TD>
      <TD>5</TD>
      <TD>5</TD>
      <TD><div style="background-color:#377EB8 !important;width:80px;height:20px;"></div></TD>
    </TR>
    <TR>
      <TD>tortosity|5 OSDR|3 OSDR</TD>
      <TD>23</TD>
      <TD>23</TD>
      <TD><div style="background-color:#4DAF4A !important;width:80px;height:20px;"></div></TD>
    </TR>
  </TBODY>
</TABLE></div>
          </div>
 The overlaps between these lists are shown in a <a href="http://circos.ca">Circos</a><sup>3</sup> plot (Figure 2.a).  Another useful representation is to overlap genes based on their functions or shared pathways.  The overlaps between gene lists can be significantly improved by considering overlaps between genes sharing the same enriched ontology term(s) (Figure 2.b).  Only ontology terms that contain less than 100 genes are used to calculate functional overlaps to avoid linking genes using very general annotation. (We do not want to link all genes, only genes that belong to specific biological processes.)
                      <div class="panel panel-info">
                        <div class="panel-heading">Figure 2. Overlap between gene lists: (a) only at the gene level, where purple curves link identical genes; (b) including the shared term level, where blue curves link genes that belong to the same enriched ontology term.  The inner circle represents gene lists, where hits are arranged along the arc.  Genes that hit multiple lists are colored in dark orange, and genes unique to a list are shown in light orange.  The publication-quality version of the figures is included in the Zip package as a .svg file under the Overlap_circos folder (readable by popular web browsers and Adobe Illustrator).</div>
                        <div class="panel-body">

                        <table>
                            <tr>
                                <td>
                                    <img src="./Overlap_circos/CircosOverlapByGene.png" style="width:500px;">
                                </td>
                                <td>
                                    <img src="./Overlap_circos/CircosOverlapByGO.png" style="width:500px;">
                                </td>
                            </tr>
                            <tr>
                                <td align='center'>
                                    
                                <a href='./Overlap_circos/CircosOverlapByGene.svg' title='download SVG file'>
                                    <img class='link' src='icon/SVG48.png' height="48" width="48" >
                                </a>
                                </td>
                                <td align='center'>
                                    
                                    <a href='./Overlap_circos/CircosOverlapByGO.svg' title='download SVG file'>
                                        <img class='link' src='icon/SVG48.png' height="48" width="48" >
                                    </a>
                                </td>
                            </tr>
                        </table>
                        </div>
                      </div><p/>



              <h3>Pathway and Process Enrichment Analysis</h3>
              For each given gene list, pathway and process enrichment analysis have been carried out with the following ontology sources: GO Biological Processes, GO Cellular Components, GO Molecular Functions, KEGG Pathway, and WikiPathways.  All genes in the genome have been used as the enrichment background. Terms with a p-value &lt; 0.05, a minimum count of 3, and an enrichment factor &gt; 1.5 (the enrichment factor is the ratio between the observed counts and the counts expected by chance) are collected and grouped into clusters based on their membership similarities.  More specifically, p-values are calculated based on the cumulative hypergeometric distribution<sup>4</sup>, and q-values are calculated using the Benjamini-Hochberg procedure to account for multiple testings<sup>5</sup>.  Kappa scores<sup>6</sup> are used as the similarity metric when performing hierarchical clustering on the enriched terms, and sub-trees with a similarity of > 0.3 are considered a cluster.  The most statistically significant term within a cluster is chosen to represent the cluster.<p/>
              When multiple gene lists are provided, all lists are merged into one list called "_FINAL".  A term may be found enriched in several individual gene lists and/or in the _FINAL gene list, and the best p-value among them is chosen as the final p-value.  The pathway/process clusters that are found to be of interest (either shared or unique based on specific list enrichment) are used to prioritize the genes that fall into those clusters (membership is presented as 1/0 binary columns in the Excel spreadsheet).  Note that individual gene lists containing more than 3000 genes are ignored during the enrichment analysis to avoid superficial terms; this is because long gene lists are often not random and generally trigger too many terms that are not of direct relevance to the biology under study.<p/>
          <div class="panel panel-info">
          <div class="panel-heading">Table 2. Top 5 clusters with their representative enriched terms (one per cluster).  "Count" is the number of genes in the user-provided lists with membership in the given ontology term. "%" is the percentage of all of the user-provided genes that are found in the given ontology term (only input genes with at least one ontology term annotation are included in the calculation).  "Log10(P)" is the p-value in log base 10.  "Log10(q)" is the multi-test adjusted p-value in log base 10.  __PATTERN__ shows the color code used for the gene lists where the term is found statistically significant, i.e., multiple colors indicate a pathway/process that is shared across multiple lists.</div>
                <div class="panel-body"><TABLE class="table">
  <THEAD>
    <TR>
      <TH class="info">_PATTERN_</TH>
      <TH class="info">GO</TH>
      <TH class="info">Category</TH>
      <TH class="info">Description</TH>
      <TH class="info">Count</TH>
      <TH class="info">%</TH>
      <TH class="info">Log10(P)</TH>
      <TH class="info">Log10(q)</TH>
    </TR>
  </THEAD>
  <TBODY>
    <TR>
      <TD><div style="background-color:#FFFFFF !important;width:20px;height:20px;float:left;"></div> <div style="background-color:#FFFFFF !important;width:20px;height:20px;float:left;"></div> <div style="background-color:#4DAF4A !important;width:20px;height:20px;float:left;"></div></TD>
      <TD>GO:1901615</TD>
      <TD>GO Biological Processes</TD>
      <TD>organic hydroxy compound metabolic process</TD>
      <TD>4</TD>
      <TD>12.12</TD>
      <TD>-3.10</TD>
      <TD>0.00</TD>
    </TR>
    <TR>
      <TD><div style="background-color:#FFFFFF !important;width:20px;height:20px;float:left;"></div> <div style="background-color:#FFFFFF !important;width:20px;height:20px;float:left;"></div> <div style="background-color:#4DAF4A !important;width:20px;height:20px;float:left;"></div></TD>
      <TD>GO:0072524</TD>
      <TD>GO Biological Processes</TD>
      <TD>pyridine-containing compound metabolic process</TD>
      <TD>3</TD>
      <TD>9.09</TD>
      <TD>-3.03</TD>
      <TD>0.00</TD>
    </TR>
    <TR>
      <TD><div style="background-color:#FFFFFF !important;width:20px;height:20px;float:left;"></div> <div style="background-color:#FFFFFF !important;width:20px;height:20px;float:left;"></div> <div style="background-color:#4DAF4A !important;width:20px;height:20px;float:left;"></div></TD>
      <TD>GO:0016757</TD>
      <TD>GO Molecular Functions</TD>
      <TD>glycosyltransferase activity</TD>
      <TD>5</TD>
      <TD>15.15</TD>
      <TD>-2.51</TD>
      <TD>0.00</TD>
    </TR>
    <TR>
      <TD><div style="background-color:#FFFFFF !important;width:20px;height:20px;float:left;"></div> <div style="background-color:#FFFFFF !important;width:20px;height:20px;float:left;"></div> <div style="background-color:#4DAF4A !important;width:20px;height:20px;float:left;"></div></TD>
      <TD>GO:0140513</TD>
      <TD>GO Cellular Components</TD>
      <TD>nuclear protein-containing complex</TD>
      <TD>4</TD>
      <TD>12.12</TD>
      <TD>-2.16</TD>
      <TD>0.00</TD>
    </TR>
    <TR>
      <TD><div style="background-color:#FFFFFF !important;width:20px;height:20px;float:left;"></div> <div style="background-color:#FFFFFF !important;width:20px;height:20px;float:left;"></div> <div style="background-color:#4DAF4A !important;width:20px;height:20px;float:left;"></div></TD>
      <TD>GO:0098796</TD>
      <TD>GO Cellular Components</TD>
      <TD>membrane protein complex</TD>
      <TD>3</TD>
      <TD>9.09</TD>
      <TD>-1.50</TD>
      <TD>0.00</TD>
    </TR>
  </TBODY>
</TABLE></div>
              </div><p/>
                  To further capture the relationships between the terms, a subset of enriched terms has been selected and rendered as a network plot, where terms with a similarity &gt; 0.3 are connected by edges.  We select the terms with the best p-values from each of the 20 clusters, with the constraint that there are no more than 15 terms per cluster and no more than 250 terms in total.  The network is visualized using <a href="http://www.cytoscape.org">Cytoscape</a><sup>7</sup>, where each node represents an enriched term and is colored first by its cluster ID (Figure 3.a) and then by its p-value (Figure 3.b).  These networks can be interactively viewed in Cytoscape through the .cys files (contained in the Zip package, which also contains a publication-quality version as a PDF) or within a browser by clicking on the web icon.  For clarity, term labels are only shown for one term per cluster, so it is recommended to use Cytoscape or a browser to visualize the network in order to inspect all node labels.  We can also export the network into a PDF file within Cytoscape, and then edit the labels using Adobe Illustrator for publication purposes.  To switch off all labels, delete the "Label" mapping under the "Style" tab within Cytoscape, and then export the network view.<p/>
                  <div class="panel panel-info">
                    <div class="panel-heading">Figure 3. Network of enriched terms: (a) colored by cluster ID, where nodes that share the same cluster ID are typically close to each other; (b) colored by p-value, where terms containing more genes tend to have a more significant p-value.</div>
                    <div class="panel-body">
                    <table>
                        <tr>
                            <td>
                                <img src="./Enrichment_GO/ColorByCluster.png" style="width:500px;">
                            </td>
                            <td>
                                <img src="./Enrichment_GO/ColorByPValue.png" style="width:500px;">
                            </td>
                        </tr>
                        <tr>
                            <td align='center'>
                                <a href='./Enrichment_GO/ColorByCluster.pdf' title='download PDF file'>
                        <img class='link' src='icon/PDF48.png' >
                        </a>
                     
                        <a href='./Enrichment_GO/GONetwork.cys' title='download CYS file'>
                        <img class='link' src='icon/CYS48.png' >
                        </a>
                    
                        
                    
                        
                        <a href='Enrichment_GO/GONetwork.html?Network=GONetwork&Style=ColorByCluster' title='interactive cytoscape' target='_blank'>
                        <img class='link' src='icon/WEB_CYS48.png' >
                        </a>
                        
                    
                            </td>
                            <td align='center'>
                                <a href='./Enrichment_GO/ColorByPValue.pdf' title='download PDF file'>
                        <img class='link' src='icon/PDF48.png' >
                        </a>
                     
                        <a href='./Enrichment_GO/GONetwork.cys' title='download CYS file'>
                        <img class='link' src='icon/CYS48.png' >
                        </a>
                    
                        
                    
                        
                        <a href='Enrichment_GO/GONetwork.html?Network=GONetwork&Style=ColorByPValue' title='interactive cytoscape' target='_blank'>
                        <img class='link' src='icon/WEB_CYS48.png' >
                        </a>
                        
                    
                            </td>
                        </tr>
                    </table>
                    </div>

                  </div><p/>
               In the case of when multiple gene lists are provided, the nodes are represented as pie charts, where the size of a pie is proportional to the total number of hits that fall into that specific term.  The pie charts are color-coded based on the gene list identities, where the size of a slice represents the percentage of genes under the term that originated from the corresponding gene list.  This plot is particularly useful for visualizing whether the terms are shared by multiple lists or unique to a specific list, as well as for understanding how these terms associate with each other within the biological context of the meta study (Figure 4).<p/>
                      <div class="panel panel-info">
                        <div class="panel-heading">Figure 4. Network of enriched terms represented as pie charts, where pies are color-coded based on the identities of the gene lists.</div>
                        <div class="panel-body">
                        <table>
                            <tr>
                                <td>
                                    <img src="./Enrichment_GO/ColorByCounts.png" style="width:1000px;">
                                </td>
                            </tr>
                            <tr>
                                <td align='center'>
                                    <a href='./Enrichment_GO/ColorByCounts.pdf' title='download PDF file'>
                        <img class='link' src='icon/PDF48.png' >
                        </a>
                     
                        <a href='./Enrichment_GO/GONetwork.cys' title='download CYS file'>
                        <img class='link' src='icon/CYS48.png' >
                        </a>
                    
                        
                    
                        
                        <a href='Enrichment_GO/GONetwork.html?Network=GONetwork&Style=ColorByCounts' title='interactive cytoscape' target='_blank'>
                        <img class='link' src='icon/WEB_CYS48.png' >
                        </a>
                        
                    
                                </td>
                            </tr>
                        </table>
                        </div>
                      </div><p/>
<h3>Protein-protein Interaction Enrichment Analysis</h3>
For each given gene list, protein-protein interaction enrichment analysis has been carried out with the following databases: STRING<sup>8</sup>, BioGrid<sup>9</sup>, OmniPath<sup>10</sup>, InWeb_IM<sup>11</sup>.All interactions in STRING are used (<a href="http://metascape.org/blog/?p=219">details</a>).  The resultant network contains the subset of proteins that form physical interactions with at least one other member in the list.  If the network contains between 3 and 500 proteins, the Molecular Complex Detection (MCODE) algorithm<sup>12</sup> has been applied to identify densely connected network components.  The MCODE networks identified for individual gene lists have been gathered and are shown in Figure 5.<p/>

                    Pathway and process enrichment analysis has been applied to each MCODE component independently, and the three best-scoring terms by p-value have been retained as the functional description of the corresponding components, shown in the tables underneath corresponding network plots within Figure 5.<p/>

      <div class="panel panel-info">
        <div class="panel-heading">Figure 5. Protein-protein interaction network and MCODE components identified in the gene lists.</div>
        <div class="panel-body">
        
                <table border-collapse='collapse'>
                <tr><td><img src="./Enrichment_PPI/tortosity|5 OSDR|3 OSDR_PPIColorByCluster.png" style="width:500px;"></td></tr>
                <tr><td align="center"><a href='./Enrichment_PPI/tortosity|5 OSDR|3 OSDR_PPIColorByCluster.pdf' title='download PDF file'>
                        <img class='link' src='icon/PDF48.png' >
                        </a>
                     
                        <a href='./Enrichment_PPI/MCODE_PPI.cys' title='download CYS file'>
                        <img class='link' src='icon/CYS48.png' >
                        </a>
                    
                        
                    
                        
                        <a href='Enrichment_PPI/PPINetwork.html?Network=tortosity|5 OSDR|3 OSDR_PPIColorByCluster&Style=PPIColorByClusterNoLabel&isPPI=True' title='interactive cytoscape' target='_blank'>
                        <img class='link' src='icon/WEB_CYS48.png' >
                        </a>
                        
                    </td></tr>
                <tr><td align="center" valign="top"><h4>tortosity|5 OSDR|3 OSDR (Full Connection) </h4></td></tr>
                </table>
                <table border-collapse='collapse'>
                <tr><td><img src="./Enrichment_PPI/_FINAL_PPIColorByCounts.png" style="width:500px;"></td></tr>
                <tr><td align="center"><a href='./Enrichment_PPI/_FINAL_PPIColorByCounts.pdf' title='download PDF file'>
                        <img class='link' src='icon/PDF48.png' >
                        </a>
                     
                        <a href='./Enrichment_PPI/MCODE_PPI.cys' title='download CYS file'>
                        <img class='link' src='icon/CYS48.png' >
                        </a>
                    
                        
                    
                        
                        <a href='Enrichment_PPI/PPINetwork.html?Network=_FINAL_PPIColorByCluster&Style=PPIColorByCountsNoLabel&isPPI=True' title='interactive cytoscape' target='_blank'>
                        <img class='link' src='icon/WEB_CYS48.png' >
                        </a>
                        
                    </td></tr>
                <tr><td align="center" valign="top"><h4>All lists merged  Colored by Counts(Full Connection) </h4><TABLE class="table">
  <THEAD>
    <TR>
      <TH class="info">GO</TH>
      <TH class="info">Description</TH>
      <TH class="info">Log10(P)</TH>
    </TR>
  </THEAD>
  <TBODY>
    <TR>
      <TD>GO:0042537</TD>
      <TD>benzene-containing compound metabolic process</TD>
      <TD>-4.7</TD>
    </TR>
    <TR>
      <TD>GO:0009723</TD>
      <TD>response to ethylene</TD>
      <TD>-3.0</TD>
    </TR>
    <TR>
      <TD>GO:0046527</TD>
      <TD>glucosyltransferase activity</TD>
      <TD>-2.9</TD>
    </TR>
  </TBODY>
</TABLE></td></tr>
                </table>
                <table border-collapse='collapse'>
                <tr><td><img src="./Enrichment_PPI/_FINAL_PPIColorByCluster.png" style="width:500px;"></td></tr>
                <tr><td align="center"><a href='./Enrichment_PPI/_FINAL_PPIColorByCluster.pdf' title='download PDF file'>
                        <img class='link' src='icon/PDF48.png' >
                        </a>
                     
                        <a href='./Enrichment_PPI/MCODE_PPI.cys' title='download CYS file'>
                        <img class='link' src='icon/CYS48.png' >
                        </a>
                    
                        
                    
                        
                        <a href='Enrichment_PPI/PPINetwork.html?Network=_FINAL_PPIColorByCluster&Style=PPIColorByClusterNoLabel&isPPI=True' title='interactive cytoscape' target='_blank'>
                        <img class='link' src='icon/WEB_CYS48.png' >
                        </a>
                        
                    </td></tr>
                <tr><td align="center" valign="top"><h4>All lists merged  Colored by Cluster(Full Connection) </h4><TABLE class="table">
  <THEAD>
    <TR>
      <TH class="info">GO</TH>
      <TH class="info">Description</TH>
      <TH class="info">Log10(P)</TH>
    </TR>
  </THEAD>
  <TBODY>
    <TR>
      <TD>GO:0042537</TD>
      <TD>benzene-containing compound metabolic process</TD>
      <TD>-4.7</TD>
    </TR>
    <TR>
      <TD>GO:0009723</TD>
      <TD>response to ethylene</TD>
      <TD>-3.0</TD>
    </TR>
    <TR>
      <TD>GO:0046527</TD>
      <TD>glucosyltransferase activity</TD>
      <TD>-2.9</TD>
    </TR>
  </TBODY>
</TABLE></td></tr>
                </table>
        </div>
      </div><p/>


    <h3>Reference</h3></p>
    <ol style="list-style: decimal inside;">
      <li>Zhou et al., Metascape provides a biologist-oriented resource for the analysis of systems-level datasets. Nature Communications (2019) 10(1):1523.</li>
      <li>Saldanha AJ. Java Treeview - extensible visualization of microarray data. Bioinformatics (2004) 20:3246-3248</li>
      <li>Krzywinski M. et al. Circos: an information aesthetic for comparative genomics. Genome Res (2009) 19:1639-1645</li>
      <li>Zar, J.H. Biostatistical Analysis 1999 4th edn., NJ Prentice Hall, pp. 523</li>
      <li>Hochberg Y., Benjamini Y. More powerful procedures for multiple significance testing. Statistics in Medicine (1990) 9:811-818.</li>
      <li>Cohen, J. A coefficient of agreement for nominal scales. Educ. Psychol. Meas. (1960) 20:27-46.</li>
      <li>Shannon P. et al., Cytoscape: a software environment for integrated models of biomolecular interaction networks. Genome Res (2003) 11:2498-2504.</li>
      <li>Szklarczyk D. et al. STRING v11: protein-protein association networks with increased coverage, supporting functional discovery in genome-wide experimental datasets. Nucleic Acids Res. (2019) 47:D607-613.</li>
      <li>Stark C. et al. BioGRID: a general repository for interaction datasets. Nucleic Acids Res. (2006) 34:D535-539.</li>
      <li>Turei D. et al. A scored human protein-protein interaction network to catalyze genomic interpretation. Nat. Methods. (2016) 13:966-967.</li>
      <li>Li T. et al. A scored human protein-protein interaction network to catalyze genomic interpretation. Nat. Methods. (2017) 14:61-64.</li>
      <li>Bader, G.D. et al. An automated method for finding molecular complexes in large protein interaction networks. BMC bioinformatics (2003) 4:2.</li>
</ol>
</div>

</body></html>